core/backend/src/test/imodel/IModel.test.ts

```diff
@@ -2956,4 +2956,101 @@ describe("iModel", () => {
 
     imodel.close();
   });
+
+  function assertSchemaVersion(imodel: IModelDb, schemaName: string, expectedVersion: string) {
+    const schemaProps = imodel.getSchemaProps(schemaName);
+    assert.isDefined(schemaProps);
+    assert.strictEqual(schemaProps.version, expectedVersion);
+  }
+
+  async function assertClassProperty(imodel: IModelDb, classNameFullName: string, properties: any) {
+    const testClass = await imodel.schemaContext.getSchemaItem(classNameFullName, EntityClass);
+    assert.isDefined(testClass);
+    for (const prop of properties) {
+      const property = await testClass?.getProperty(prop.propName);
+      assert.equal(prop.propertyExists, property !== undefined, `Property ${prop.propName} existence check failed for class ${classNameFullName}`);
+    }
+  }
+
+  it("Major schema updates should only work for dynamic schemas", async () => {
+    const imodelPath = IModelTestUtils.prepareOutputFile("IModel", "majorschemaupgrade.bim");
+
+    // Remove the file if it already exists
+    if (IModelJsFs.existsSync(imodelPath))
+      IModelJsFs.unlinkSync(imodelPath);
+
+    // Create a new empty snapshot iModel
+    const testImodel = SnapshotDb.createEmpty(imodelPath, { rootSubject: { name: "majorschemaupgrade" } });
+
+    // Import initial schema and verify version
+    await testImodel.importSchemaStrings([`<?xml version="1.0" encoding="utf-8" ?>
+      <ECSchema schemaName="TestSchema" alias="ts" version="1.0.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+          <ECSchemaReference name="BisCore" version="1.0.0" alias="bis"/>
+
```

Comment:
You are missing and second is dynamic schema are allowed to have delete without requiring change to major version.     
```xml
   <ECCustomAttributes>
        <DynamicSchema xmlns="CoreCustomAttributes.01.00.04"/>
    </ECCustomAttributes>
```